### PR TITLE
fix: Force close after updates

### PR DESCRIPTION
**Pull request by D4nte**
I found a couple of bugs :)

### CODE REVIEW

File: thor/tests/e2e.rs [hunk 1]
```diff
@@ -253,3 +253,145 @@ async fn e2e_force_close_channel() {
...
+        Balance {
+            ours: expected_bob_balance,
+            theirs: expected_alice_balance,
+        },
+        time_lock,
+    );
+
+    futures::future::try_join(alice_update, bob_update)
+        .await
+        .unwrap();
```
Comment (bonomat) [inline thor/tests/e2e.rs:358]: Why do you do 2 updates?

**Review by bonomat — APPROVED** [review]
Good catch.

File: thor/tests/e2e.rs (hunk 1 above)
Comment (D4nte) [reply thor/tests/e2e.rs:358]: To be sure :)

File: thor/src/protocols/create.rs [hunk 2]
```diff
@@ -244,10 +244,17 @@ impl State2 {
...
-            (half_amount, self.final_address_self.clone()),
-            (half_amount, self.final_address_other.clone()),
+            (
+                self.balance.ours - fee_per_participant,
+                self.final_address_self.clone(),
+            ),
+            (
+                self.balance.theirs - fee_per_participant,
+                self.final_address_other.clone(),
+            ),
```
[the remark this answers is not in the document]
Comment (D4nte) [reply thor/src/protocols/create.rs]: I don't understand this comment, below here?

[the remark this answers is not in the document]
Comment (luckysori) [reply thor/src/protocols/create.rs]: By below I meant in the other place where we call this function (update protocol). Sorry for the confusion.

File: thor/src/transaction.rs [hunk 3]
```diff
@@ -436,16 +443,19 @@ impl SplitTransaction {
...
-        let half_fee = TX_FEE / 2;
+        // Distribute transaction TX_c fee costs evenly between outputs
+        let half_TX_c_fee = TX_c.fee() / 2;
+
+        // Distribute transaction TX_s fee costs evenly between outputs
+        let half_TX_s_fee = TX_s_fee / 2;
 
         let output_0 = TxOut {
-            value: amount_0.as_sat() - half_fee,
+            value: amount_0.as_sat() - half_TX_c_fee.as_sat() - half_TX_s_fee.as_sat(),
```
Comment (D4nte) [inline thor/src/transaction.rs:453]: That's a typo for sure.

Comment (D4nte) [reply thor/src/transaction.rs:453]: @luckysori

Comment (luckysori) [reply thor/src/transaction.rs:453]: If you're referring to `half_TX_c_fee`, it's not. If you pass in the amount that each party puts in the channel (in this case `amount_0`), you have to subtract both the party's fee contribution for `TX_c` and for `TX_s`.

Comment (D4nte) [reply thor/src/transaction.rs:453]: Ah well, this is the issue with having mono-letter variables (or close to).

Comment (luckysori) [reply thor/src/transaction.rs:453]: We can change that. I love long names.

**Comment by D4nte** [comment]
@luckysori I am happy to merge with your changes 😉

**Pull request merged**